multi-transfer-esdt/src/lib.rs

```diff
@@ -48,41 +33,53 @@ pub trait MultiTransferEsdt {
         &self,
         #[var_args] transfers: VarArgs<SingleTransferTuple<Self::BigUint>>,
     ) -> SCResult<MultiResultVec<TransactionStatus>> {
-        only_owner!(self, "only owner may call this function");
+        self.require_caller_owner()?;
 
         let mut tx_statuses = Vec::new();
+        let mut cached_token_ids = Vec::new();
+        let mut cached_prices = Vec::new();
 
         for (i, (to, token_id, amount)) in transfers.into_vec().iter().enumerate() {
             if to.is_zero() || self.blockchain().is_smart_contract(to) {
                 tx_statuses.push(TransactionStatus::Rejected);
                 continue;
             }
-            if !self.token_whitelist().contains(token_id) || !self.is_local_mint_role_set(token_id)
+            if !self.token_whitelist().contains(token_id) || !self.is_local_role_set(token_id, &EsdtLocalRole::Mint)
             {
                 tx_statuses.push(TransactionStatus::Rejected);
                 continue;
             }
 
-            self.send().esdt_local_mint(token_id, amount);
-            self.send().direct(to, token_id, amount, &[i as u8]);
+            let queried_fee: Self::BigUint;
+            let required_fee = match cached_token_ids.iter().position(|&id| id == token_id) {
+                Some(index) => &cached_prices[index],
```

Comment:
when is the cached price invalidated ?

Reply:
I think we should always query the fee.

Reply:
The "cache" is a local variable, so this is per-batch only.
There is no need to query again during the same endpoint execution, as that will bring the same price every single time.